<?xml version="1.0" ?>
<launch>
  <!-- This launch files runs the pipeline for a camera that has already started up -->

  <!-- Mode of object detector -->
  <arg name="continuous_streaming"		default="true"/>
  <arg name="camera_name"	default="b_bot_outside_camera"/>

  <!-- Localization parameters -->
  <arg name="nposes"		default="2"/>
  <arg name="timeout"		default="10"/>
  <arg name="ply_dir"		default="$(find o2ac_parts_description)/meshes"/>
  <arg name="urdf_dir"		default="$(find o2ac_parts_description)/urdf/generated"/>

  <!-- Construct recognition pipeline -->
  <include unless="$(arg continuous_streaming)" file="$(find o2ac_vision)/launch/o2ac_recognition.launch">
    <arg name="nposes"		  value="$(arg nposes)"/>
    <arg name="timeout"		  value="$(arg timeout)"/>
    <arg name="ply_dir"		  value="$(arg ply_dir)"/>
    <arg name="urdf_dir"	  value="$(arg urdf_dir)"/>
    <arg name="camera_info_topic" value="$(arg camera_name)/color/camera_info"/>
    <arg name="image_topic"	  value="$(arg camera_name)/color/image_raw"/>
    <arg name="depth_topic"	  value="$(arg camera_name)/aligned_depth_to_color/image_raw"/>
    <arg name="normal_topic"	  value=""/>
  </include>

  <node if="$(arg continuous_streaming)" name="object_detector" pkg="o2ac_vision" type="o2ac_vision_server.py" output="screen">
    <remap from="/image" to="$(arg camera_name)/color/image_raw"/>
    <param name="continuous_streaming" value="true"/>
  </node>

  <!-- Spawn visualizer and parameter configurator -->
  <node name="$(anon rviz)" pkg="rviz" type="rviz" output="screen"
	args="-d $(find o2ac_vision)/launch/realsense.rviz"/>
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>

</launch>
